src/modules.ts

```diff
@@ -0,0 +1,30 @@
+let currentModule: string | null = null;
+export function setCurrentModule(mod: string | null) {
+  currentModule = mod;
+}
+
+type CleanupFn = () => void | Promise<void>;
+let cleanupFns: Array<{ fn: CleanupFn, module: string }> = [];
+
+export function registerModuleCleanup(fn: () => void | Promise<void>) {
+  if (!currentModule) {
+    // TODO: print warning? or collect cleanup that do not belong to a module?
```

Comment:
this should be implemented? at least a warning sounds like a good idea

Reply:
done